### PR TITLE
finished week7

### PR DESCRIPTION
**Pull request by vlad-pisotskyi**
featured, company, name, sort, fields, numericFilters received from client to server. implemented find() select() sort() skip() limit() methods on moonguse models. Discovered new mongoose model properties like default, required, enums etc. used regex to 'translate' query to mongoose

### CODE REVIEW

File: 04-store-api/starter/controllers/products.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const Product = require("../models/product");
+require("colors");
```
Comment (a-zuniga) [inline 04-store-api/starter/controllers/products.js:2]: I don't see this being used anywhere, unused import?

File: 04-store-api/starter/models/product.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		default: Date.now(),
+	},
+	// mongoose will not allow any other values
+	// than on of enum array values
+	// company: {
+	// 	type: String,
+	// 	enum: ["ikea", "liddy", "caressa", "marcos"],
+	// },
+
+	// same with error handling
```
Comment (a-zuniga) [inline 04-store-api/starter/models/product.js]: nitpick but may not need the comment here. When defining a schema with specific values, any schema validation check will fail on values not defined in `values`

File: 04-store-api/starter/app.js [hunk 3]
```diff
@@ -1 +1,38 @@
...
+// product route
+
+app.use(notFoundMiddleware);
+app.use(errorMiddleware);
+
+const port = process.env.PORT || 3000;
+
+const start = async () => {
+	try {
+		await connectDB(process.env.MONGO_URI);
```
Comment (a-zuniga) [inline 04-store-api/starter/app.js:32]: In order to use `await` on `connectDB()` it needs to be able to return a promise.

File: 04-store-api/starter/db/connect.js [hunk 4]
```diff
@@ -1,12 +1,7 @@
-const mongoose = require('mongoose')
+const mongoose = require("mongoose");
 
 const connectDB = (url) => {
-  return mongoose.connect(url, {
```
Comment (a-zuniga) [inline 04-store-api/starter/db/connect.js:4]: In order to use `await` on this inside `app.js` you need to return the call to `mongoose.connect()`. I wouldn't do away with this change!

Comment (a-zuniga) [reply 04-store-api/starter/db/connect.js:4]: For context, this will ensure that you return a promise back to the function that calls `connectDB()`.

**Review by a-zuniga — APPROVED** [review]
Left some comments, mainly nitpicks. Going to approve for now but lets sync up and work through some of those comments I left.

File: 04-store-api/starter/controllers/products.js (hunk 1 above)
Comment (vlad-pisotskyi) [reply 04-store-api/starter/controllers/products.js:2]: i'm sorry i'm using it for debugging sometimes. it paints console logs in the terminal. i forgot to delete it

File: 04-store-api/starter/app.js (hunk 3 above)
Comment (vlad-pisotskyi) [reply 04-store-api/starter/app.js:32]: i think mongoose.connect is a promise https://mongoosejs.com/docs/connections.html

**Pull request merged**